### DISCUSSION TITLE
Build of v1.21.0 does not work with BLE in Pico W

### DISCUSSION THREAD

**Discussion by robofwy** [micropython/micropython#12938] [RP2040 / Pico]
I have managed to build v1.21.0 (on a 32-bit Debian host) for the RPI_PICO_W board and everything seems to work except Bluetooth/BLE which locks it up. The firmware uf2 file is a different size to the official binary release so I am not sure what is wrong there.

picotool shows the same output for both the official binary and my built one. There are no obvious errors in the build process.

The board locks up on a scan using aioble.scan(). I guess this could be BLE related or uasyncio related since none of my other tests use async code.

If I build the latest v1.22 master/preview, everything (including BLE/uasyncio) seems to work.

**Comment by jimmo** [comment]
This is very likely to be the same as https://github.com/micropython/micropython/issues/12617 which explains why it's fixed with the latest version.

Fortunately we don't seem to be seeing the issue on the release version (different compiler version on the build server likely).

**Reply by robofwy** [reply]
It is indeed the same as #12617 - I did search existing issues before posting but somehow missed that one!

I can confirm the following all (independently) fix the problem:

Building (unpatched) v1.21.0 with ARM gcc 12.3.1 or 13.2.1 (manually downloaded and added to head of PATH)
Building (unpatched) v1.21.0 with ARM gcc 12.2.1 and increasing the stack size from 4096 to 5120 in boards/RPI_PICO_W/mpconfigboard.cmake
Building v1.21.0 with patch #12647 with ARM gcc 12.2.1

Of the above, I am less keen on the first two as there could be some situations outside of the tests that cause the stack to exceed the limit. The first. in particular. may be "sailing very close to the wind" and may explain any lock-ups observed with the official binary release of 1.21.0 when using Bluetooth.
